.github/workflows/on-push-build-with-cache.yml

```diff
@@ -0,0 +1,38 @@
+name: On Push Build with Cache
+on:
+  push:
+    branches:
+      - "*"
+
+jobs:
+  docker:
+    name: Build Docker with Cache
+    runs-on: ubuntu-20.04
+    environment: production
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+
+      - name: Determine Meta
+        run: echo "REF=${GITHUB_REF##*/}" >> $GITHUB_ENV
+
+      - name: Dockerhub Login
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Docker Build and Push (with cache)
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          cache-from: type=registry,ref=${{ secrets.DOCKER_REPOSITORY }}:${{ env.REF }}-cache
+          cache-to: type=registry,ref=${{ secrets.DOCKER_REPOSITORY }}:${{ env.REF }}-cache,mode=max
+          push: true
+          build-args: |
+            GIT_REPO=https://github.com/${{ github.repository }}.git
+            GIT_REF=${{ env.REF }}
+          tags: ${{ secrets.DOCKER_REPOSITORY }}:${{ env.REF }}
```

Comment:
Is there some particular reason to store the docker repository in a repo secret?